### DISCUSSION TITLE
Can't install HomeBrew on Mac M1

### DISCUSSION THREAD

**Discussion by Jbat1023** [Homebrew/discussions#3136] [Getting started]
Hello,

Trying to install Homebrew on my MBP M1 and get the following error message
`xcrun: error: unable to load libxcrun (dlopen(/Library/Developer/CommandLineTools/usr/lib/libxcrun.dylib, 0x0005): tried: '/Library/Developer/CommandLineTools/usr/lib/libxcrun.dylib' (mach-o file, but is an incompatible architecture (have 'x86_64', need 'arm64e')), '/usr/lib/libxcrun.dylib' (no such file)).
Failed during: git init -q`

**Comment by gromgit** [comment]
See #3062

**Comment by Mr-Kary** [comment]
I have a different error installing homebrew on MBP M1 Monterey 12.2

[LoranKaomputer3:/opt] kary(42)->/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
Illegal variable name.

I am wanting to use homebrew to install php but I see some are having issues with that too.

Any suggestions?

**Reply by gromgit** [reply]
Sounds like you're using a shell that doesn't understand the `$()` construct. Try running this instead:
```
curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh | /bin/bash
```
It will issue the following warning:
```
Warning: Running in non-interactive mode because `stdin` is not a TTY.
```
but should otherwise work normally.

**Reply by Mr-Kary** [reply]
Thank you that was very helpful.  I was able to install homebrew and use it to install php.  But then a new problem -- apache will not load the php module  /opt/homebrew/opt/php/lib/httpd/modules/libphp.so because code signing is absent.

**Reply by carlocab** [reply]
What is the output of `brew config` and `brew doctor`?

**Reply by Mr-Kary** [reply]
But then I found this -- https://www.simplified.guide/macos/apache-php-homebrew-codesign  and now it looks like I am all good
